src/main/java/com/github/manolo8/darkbot/gui/utils/FloatingDialog.java

```diff
@@ -1,22 +1,21 @@
 package com.github.manolo8.darkbot.gui.utils;
 
-import javax.swing.*;
-import javax.swing.border.Border;
-import java.awt.*;
-import java.awt.event.FocusEvent;
-import java.awt.event.FocusListener;
-import java.awt.event.WindowAdapter;
-import java.awt.event.WindowEvent;
-import java.beans.PropertyChangeEvent;
-import java.beans.PropertyChangeListener;
+import javax.swing.BorderFactory;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.JPopupMenu;
 
 public class FloatingDialog {
 
     public static void show(JPanel panel, JComponent parent) {
+        show(panel, parent, 0, parent.getHeight());
+    }
+
+    public static JPopupMenu show(JComponent component, JComponent parent, int x, int y) {
         JPopupMenu popup = new JPopupMenu();
         popup.setBorder(BorderFactory.createEmptyBorder());
-        popup.add(panel);
-        popup.show(parent, 0, parent.getHeight());
+        popup.add(component);
+        popup.show(parent, x, y);
+        return popup;
```

Comment:
i'd revert all changes in this, keep the FloatingDialog class for... well.. arbitrary JPanels that you want to show as floating dialogs, which was the main mission. The whole showing a jmenu is just already supported behavior that's available via the jmenu itself, which is what ended up happening anyways for the use-case you tried to migrate to use this